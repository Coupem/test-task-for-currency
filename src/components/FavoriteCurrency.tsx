// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';
import {
  FavoriteCurrencyWrapper,
  NoFavoriteCurrency,
  TitleFavoriteCurrency,
  ListGroupContainer,
} from '../styled';
import { FavoriteCurrencyProp } from '../types';

const FavoriteCurrency = ({
  favoriteCurrencies,
}: FavoriteCurrencyProp): JSX.Element => (
  <FavoriteCurrencyWrapper>
    <TitleFavoriteCurrency>Favorite Currency:</TitleFavoriteCurrency>

    {Object.values(favoriteCurrencies).length === 0 && (
      <NoFavoriteCurrency>NO CURRENCY</NoFavoriteCurrency>
    )}

    {Object.values(favoriteCurrencies).map((breakpoint) => (
      <ListGroupContainer key={breakpoint.ccy}>
        <ListGroup.Item>Currency: {breakpoint.ccy}</ListGroup.Item>
        <ListGroup.Item>Buy: {breakpoint.buy}</ListGroup.Item>
        <ListGroup.Item>Sale: {breakpoint.sale}</ListGroup.Item>
      </ListGroupContainer>
    ))}
  </FavoriteCurrencyWrapper>
);

FavoriteCurrency.propTypes = {
  favoriteCurrencies: PropTypes.shape({
    ccy: PropTypes.string,
    buy: PropTypes.string,
    sale: PropTypes.string,
  }),
};

export default FavoriteCurrency;
