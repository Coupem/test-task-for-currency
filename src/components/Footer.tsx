import React from 'react';
import { FooterComponent } from '../styled';

const Footer = (): JSX.Element => (
  <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>
);

Footer.propTypes = {};

export default Footer;
