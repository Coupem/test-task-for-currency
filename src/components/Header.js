import React from 'react';
import { HeaderComponent } from '../styled';

const Header = () => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
