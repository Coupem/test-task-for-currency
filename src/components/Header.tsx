// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import { HeaderComponent } from '../styled';

const Header = (): JSX.Element => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
