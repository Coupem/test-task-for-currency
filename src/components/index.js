export { default as InputValuesGroup } from './Converter/InputValuesGroup';
export { default as ConvertCurrency } from './Converter/ConvertCurrency';
export { default as FavoriteCurrency } from './Course/FavoriteCurrency';
export { default as Navigation } from './Navigation/Navigation';
export { default as Header } from './Header/Header';
export { default as Footer } from './Footer/Footer';
