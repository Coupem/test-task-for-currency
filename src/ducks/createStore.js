import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas/sagas';
import rootReducer from './reducers/reducer';

export default function () {
  const sagaMiddleware = createSagaMiddleware();
  const composeEnhancers =
    typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose;

  const enhancer = composeEnhancers(applyMiddleware(sagaMiddleware));
  const store = createStore(rootReducer, {}, enhancer);
  sagaMiddleware.run(rootSaga);

  return store;
}
