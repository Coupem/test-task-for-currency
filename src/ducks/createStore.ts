import { createStore, applyMiddleware, compose, Action } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas/sagas';
import rootReducer from './reducers/reducer';

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

export default function () {
  const sagaMiddleware = createSagaMiddleware();
  const composeEnhancers =
    typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()
      : compose;

  const enhancer = composeEnhancers(applyMiddleware(sagaMiddleware));
  const store = createStore(rootReducer, {}, enhancer);
  sagaMiddleware.run(rootSaga);

  return store;
}
