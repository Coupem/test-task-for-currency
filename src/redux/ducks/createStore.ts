import { createStore, applyMiddleware, compose, Store } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas';
import rootReducer from './reducer';
import { StoreState } from './types';

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

export default function (): Store<StoreState> {
  const sagaMiddleware = createSagaMiddleware();

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const enhancer = composeEnhancers(applyMiddleware(sagaMiddleware));

  const store = createStore(rootReducer, {}, enhancer);
  sagaMiddleware.run(rootSaga);

  return store;
}
