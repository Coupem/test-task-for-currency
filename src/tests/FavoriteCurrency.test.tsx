import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { Provider } from 'react-redux';
import { FavoriteCurrency } from '../components/index';
import createStore from '../redux';

describe('Favorite currency component', () => {
  const wrapper = shallow(
    <Provider store={createStore()}>
      <FavoriteCurrency />
    </Provider>
  );

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
