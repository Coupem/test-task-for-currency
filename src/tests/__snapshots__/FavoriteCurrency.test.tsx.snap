// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite currency component should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FavoriteCurrency
    BTC={
      Object {
        "buy": "31125.3298",
        "ccy": "BTC",
        "sale": "34401.6803",
      }
    }
    EUR={
      Object {
        "buy": "31.85000",
        "ccy": "EUR",
        "sale": "32.45000",
      }
    }
    RUR={
      Object {
        "buy": "0.35000",
        "ccy": "RUR",
        "sale": "0.38000",
      }
    }
    USD={
      Object {
        "buy": "27.05000",
        "ccy": "USD",
        "sale": "27.45000",
      }
    }
  />
</ContextProvider>
`;
